lib/interface_quda.cpp

```diff
@@ -3932,6 +3945,80 @@ void computeKSLinkQuda(void *fatlink, void *longlink, void *ulink, void *inlink,
   profileFatLink.TPSTOP(QUDA_PROFILE_TOTAL);
 }
 
+void computeTwoLinkQuda(void *twolink, void *inlink, QudaGaugeParam *param)
+{
+  profileGaussianSmear.TPSTART(QUDA_PROFILE_TOTAL);
+  profileGaussianSmear.TPSTART(QUDA_PROFILE_INIT);
+
+  checkGaugeParam(param);
+
+  GaugeFieldParam gParam(*param, inlink, QUDA_GENERAL_LINKS);
+  gParam.gauge     = twolink;
+  cpuGaugeField cpuTwoLink(gParam);  // create the host twolink
+  profileGaussianSmear.TPSTOP(QUDA_PROFILE_INIT);
+
+  cudaGaugeField *cudaInLinkEx = nullptr;
+
+  if(inlink) {
+    gParam.link_type = param->type;
+    gParam.gauge     = inlink;
+    cpuGaugeField cpuInLink(gParam);    // create the host sitelink
+
+    // create the device fields
+    gParam.reconstruct = param->reconstruct;
+    gParam.setPrecision(param->cuda_prec, true);
+    gParam.create = QUDA_NULL_FIELD_CREATE;
+    cudaGaugeField *cudaInLink = new cudaGaugeField(gParam);
+    profileGaussianSmear.TPSTOP(QUDA_PROFILE_INIT);
+
+    cudaInLink->loadCPUField(cpuInLink, profileGaussianSmear);
+    //
+    cudaInLinkEx = createExtendedGauge(*cudaInLink, R, profileGaussianSmear);
+    //
+    profileGaussianSmear.TPSTART(QUDA_PROFILE_FREE);
+    delete cudaInLink;
+    profileGaussianSmear.TPSTOP(QUDA_PROFILE_FREE);
+
+  } else {
+    cudaInLinkEx = createExtendedGauge(*gaugePrecise, R, profileGaussianSmear);
+  }
+
+  GaugeFieldParam gsParam(*gaugePrecise);
+
+  gsParam.create        = QUDA_NULL_FIELD_CREATE;
+  gsParam.link_type     = QUDA_ASQTAD_LONG_LINKS;
+  gsParam.reconstruct   = QUDA_RECONSTRUCT_NO;
+  gsParam.setPrecision(param->cuda_prec, true);
+  gsParam.ghostExchange = QUDA_GHOST_EXCHANGE_PAD;
+  gsParam.nFace         = 3;
+  gsParam.pad           = gsParam.pad*gsParam.nFace;
+
+  profileGaussianSmear.TPSTART(QUDA_PROFILE_INIT);
+
+  if(gaugeSmeared != nullptr) delete gaugeSmeared;
+  gaugeSmeared = new cudaGaugeField(gsParam);
+
+  
+  profileGaussianSmear.TPSTOP(QUDA_PROFILE_INIT);
+
+  profileGaussianSmear.TPSTART(QUDA_PROFILE_COMPUTE);
+
+  computeTwoLink(*gaugeSmeared, *cudaInLinkEx);
+  gaugeSmeared->exchangeGhost();
+
+  profileGaussianSmear.TPSTOP(QUDA_PROFILE_COMPUTE);
+  //
+  gaugeSmeared->saveCPUField(cpuTwoLink, profileGaussianSmear);
+
+  profileGaussianSmear.TPSTART(QUDA_PROFILE_FREE);
+  
+  delete gaugeSmeared;gaugeSmeared = nullptr;
```

Comment:
that should not be a single line but I think we need a clang-format on the diff anyway

Reply:
Thanks for finding it out. Split into two lines. https://github.com/lattice/quda/commit/8e7ad58492d3a863a6af6f868dd28d5354de57e8